mantidproject/mantid
# SliceViewer: Exception when changing peak workspace in the ADS when overlaid

**Description of work.**
Adds handles for the ADS observer in the peaksviewer presenter, which handles if a workspace is removed from underneath the sliceviewer if a peaks workspace is loaded in.



**To test:**
- Run this Script in workbench:
```
from mantid.simpleapi import *

SXD23767 = Load(Filename='SXD23767.raw', LoadMonitors='Exclude')

SetUB(SXD23767, 1,1,2,90,90,120)
md_non_ortho = ConvertToDiffractionMDWorkspace(InputWorkspace='SXD23767', OutputDimensions='HKL')
```
- Follow issue instructions to create peaks workspace
- Open slice viewer on md_non_ortho
- Add peaks workspace in using this button:
![image](https://user-images.githubusercontent.com/24723609/104702015-52c89300-570d-11eb-9e0d-efd133ebe316.png)
- Ensure the peaks workspace is displayed on the data display by double clicking the workspace table display on the right of the slice viewer
- Then remove the peaks workspace from the ADS, the UI should update to reflect the new ADS state.
- Repeat but for renaming the peaks workspace, replacing the peaks workspace, and clearing the ADS as a whole.
 



Fixes #30308 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
